docs/notes/pytorch_notes.rst

```diff
@@ -121,13 +122,58 @@ activation functions.
 
 Example
 ---------
-``PyTorch`` supports inplace computation for some commonly used
+PyTorch supports inplace computation for some commonly used
 activations such as ``ReLU``, but not all of them.
 For exmaple on implementing an inplace activation function, please refer to this
 `PR on inplace Softsign <https://github.com/HorizonRobotics/alf/pull/544>`_.
 
 
 
+5.  Constructing tensor from numpy float scalar
+=================================================
+Description
+-----------------
+Numpy's default dtype is ``float64`` while PyTorch's default is ``float32``.
+When constructing a tensor from numpy scalar of dtype ``float64``,
+older versions of PyTorch (e.g. 1.4) will convert ``np.float64`` to ``torch.float32`` (`reference issue <https://github.com/pytorch/pytorch/issues/27754>`_), as shown
+in the example below:
+
+.. code-block:: python
+    # with PyTorch 1.4
+    x = torch.rand(2, 3)       # torch.float32
+    a = np.log(2)              # numpy.float64
+    a_tensor = torch.tensor(a) # torch.float32, PyTorch 1.4 does dtype conversion here
+
+    # a test function
+    torch.where(x < a, x, a_tensor) # works correctly in PyTorch 1.4
+
+The example can be executed without any issues in PyTorch 1.4.
+However, in newer version of PyTorch, this dtype conversion has been removed.
+This may break some codes that have no issues when with lower versions.
+
+For example, running the same piece of code as above in PyTorch 1.8,
+``a_tensor`` now has a dtype of ``torch.float64``, which breaks the test
+function ``torch.where(x < a, x, a_tensor)``, since now ``x`` and ``a_tensor``
+have different dtypes.
+
+
+
+6. Type promotion
+=====================
+Description
+-----------------
+PyTorch 1.8 has auto type promotion during computation, which is different from
+PyTorch 1.4.
+
+For example:
 
+.. code-block:: python
+    a = torch.tensor(5, dtype=torch.int64)
+    b = a / 2
```

Comment:
Please also note that even if a is divisible by 2 or a is an array, the dtype of the result will be float.

Reply:
Updated the doc.